packages/common/src/compose-tunnel-agent.ts

```diff
@@ -18,3 +18,55 @@ export type ScriptInjection = {
   defer?: boolean
   async?: boolean
 }
+
+type Stringified<T> = {
+  [k in keyof T]: string
+}
+const parseScriptInjection = ({ pathRegex, defer, async, src }: Stringified<ScriptInjection>):
+ ScriptInjection | Error => {
+  try {
+    if (!src) {
+      throw new Error('missing src')
+    }
+    return {
+      ...pathRegex && { pathRegex: new RegExp(pathRegex) },
+      ...defer && { defer: defer === 'true' },
+      ...async && { async: async === 'true' },
+      src,
+    }
+  } catch (e) {
+    return e as Error
+  }
+}
+
+export const extractSectionsFromLabels = <T>(prefix: string, labels: Record<string, string>) => {
+  const re = new RegExp(`^${prefix.replace(/\./g, '\\.')}\\.(?<id>.+?)\\.(?<key>[^.]+)$`)
+  const sections:{[id:string]: T } = {}
+  for (const [label, value] of Object.entries(labels)) {
+    const match = label.match(re)?.groups
+    if (match) {
+      set(sections, [match.id, camelCase(match.key)], value)
```

Comment:
Ideally, it'd be parseScriptInjection = recordParser(def)
// recordParser = (def<T>) => (Record<string,string >)=> T
// where def<T> can be something declarative as Zod)

But I think it's a bit pre-mature to extract it and Zod is a bit tricky since it translate 'false' to true which might be correct in js, but we are in the context of configuration.